provider/cldr/src/transform/list/mod.rs

```diff
@@ -141,29 +273,95 @@ impl<'a> TryFrom<&cldr_serde::list_patterns_json::ListPatterns> for ListFormatte
     }
 }
 
-#[test]
-fn test_basic() {
+#[cfg(test)]
+mod tests {
+    use super::*;
     use icu_locid_macros::langid;
+    use icu_locid::LanguageIdentifier;
 
-    let cldr_paths = crate::cldr_paths::for_test();
-    let provider = ListProvider::try_from(&cldr_paths as &dyn CldrPaths).unwrap();
-
-    let de_list: DataPayload<ListFormatterPatternsV1Marker> = provider
-        .load_payload(&DataRequest {
-            resource_path: ResourcePath {
-                key: key::LIST_FORMAT_V1,
-                options: ResourceOptions {
-                    variant: None,
-                    langid: Some(langid!("fr")),
+    fn provide<'data>(lang: LanguageIdentifier) -> DataPayload<'data, ListFormatterPatternsV1Marker> {
+        let cldr_paths = crate::cldr_paths::for_test();
+        let provider = ListProvider::try_from(&cldr_paths as &dyn CldrPaths).unwrap();
+        provider
+            .load_payload(&DataRequest {
+                resource_path: ResourcePath {
+                    key: key::LIST_FORMAT_V1,
+                    options: ResourceOptions {
+                        variant: None,
+                        langid: Some(lang),
+                    },
                 },
-            },
-        })
-        .unwrap()
-        .take_payload()
-        .unwrap();
-
-    assert_eq!(
-        de_list.get().patterns.and.wide.end,
-        ConditionalListJoinerPattern::from_str("{0} et {1}").unwrap()
-    );
+            })
+            .unwrap()
+            .take_payload()
+            .unwrap()
+    }
+
+    #[test]
+    fn test_basic() {
+        assert_eq!(
+            provide(langid!("fr")).get().patterns.and.wide.end,
+            ConditionalListJoinerPattern::from_str("{0} et {1}").unwrap()
+        );
+    }
+
+    #[test]
+    fn test_spanish() {
+        let y_parts = ("", " y ", "");
+        let e_parts = ("", " e ", "");
+        let o_parts = ("", " o ", "");
+        let u_parts = ("", " u ", "");
+
+        // ... y Mallorca
+        assert_eq!(provide(langid!("es")).get().patterns.and.wide.end.parts("Mallorca"), y_parts);
```

Comment:
Suggestion(optional):
```suggestion
        let u_parts = ("", " u ", "");
        let payload = provide(langid!("es"));
        let patterns = &payload.get().patterns;

        // ... y Mallorca
        assert_eq!(patterns.and.wide.end.parts("Mallorca"), y_parts);
```